auth/jwt/middleware.go

```diff
@@ -0,0 +1,111 @@
+package jwt
+
+import (
+	"errors"
+
+	"golang.org/x/net/context"
+
+	jwt "github.com/dgrijalva/jwt-go"
+	"github.com/go-kit/kit/endpoint"
+)
+
+const (
+	// JWTTokenContextKey holds the key used to store a JWT Token in the context
+	JWTTokenContextKey = "JWTToken"
```

Comment:
Shouldn't this (and JWTClaimsContextKey) have an unexported type to avoid collisions? Something like

```go
type key string

const (
	// JWTTokenContextKey holds the key used to store a JWT Token in the context
 	JWTTokenContextKey key = "JWTToken"
	// JWTClaimsContxtKey holds the key used to store the JWT Claims in the context
 	JWTClaimsContextKey key = "JWTClaims"
)
```

See https://blog.golang.org/context#TOC_3.2.